src/client/executor.rs

```diff
@@ -651,78 +661,9 @@ impl Client {
         let start_time = Instant::now();
         let command_result = match connection.send_message(message).await {
             Ok(response) => {
-                async fn handle_response<T: Operation>(
```

Comment:
I pulled this out and renamed it for readability, but the body is unchanged.

Reply:
ha, I also did this in #1358 (minus the rename)